src/types.ts

```diff
@@ -1,64 +1,45 @@
-import type { RequestError } from "@octokit/request-error";
-import type { Schema } from "@octokit/webhooks-definitions/schema";
-import type { EmitterEventWebhookPayloadMap } from "./generated/get-webhook-payload-type-from-event";
-
-type EmitterEventPayloadMap = { "*": Schema } & EmitterEventWebhookPayloadMap;
-
-export type EmitterWebhookEventMap = {
-  [K in keyof EmitterEventPayloadMap]: BaseWebhookEvent<K>;
-};
-
-export type EmitterWebhookEventName = keyof EmitterWebhookEventMap;
-export type EmitterWebhookEvent = EmitterWebhookEventMap[EmitterWebhookEventName];
-
-/**
- * A map of all possible emitter events to their event type.
- * AKA "if the emitter emits x, the handler will be passed y"
- */
-export type EmitterEventMap = EmitterWebhookEventMap & {
-  error: WebhookEventHandlerError;
-};
-export type EmitterEventName = keyof EmitterEventMap;
-export type EmitterEvent = EmitterEventMap[EmitterEventName];
-
-export type EmitterAnyEvent = EmitterWebhookEventMap["*"];
-
-export type ToWebhookEvent<
-  TEmitterEvent extends string
-> = TEmitterEvent extends `${infer TWebhookEvent}.${string}`
-  ? TWebhookEvent
-  : TEmitterEvent;
-
-interface BaseWebhookEvent<
-  TName extends keyof EmitterEventPayloadMap = keyof EmitterEventPayloadMap
-> {
+import { RequestError } from "@octokit/request-error";
+import type {
+  WebhookEventMap,
+  WebhookEventName,
+} from "@octokit/webhooks-definitions/schema";
+import type { emitterEventNames } from "./generated/webhook-names";
+
+export type EmitterWebhookEventName = typeof emitterEventNames[number];
+export type EmitterWebhookEvent<
+  TEmitterEvent extends EmitterWebhookEventName = EmitterWebhookEventName
```

Comment:
```suggestion
  TEmitterEvent extends EmitterWebhookEventName = WebhookEventName
```
I don't know if it makes a difference, but in #441 I went with `WebhookEventName` here because `EmitterWebhookEvent<WebhookEventName>` is functionally equivalent to `EmitterWebhookEvent<EmitterWebhookEventName>` and `WebhookEventName` is smaller than `EmitterWebhookEventName` so I thought it might perform better? https://github.com/jablko/webhooks.js/compare/patch-2..octokit:next#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aL14-R10

Reply:
I'd prefer to use the most accurate type - if it was causing more performance problems, I'd want to know about them here since you could still run into them in userland by using that type :)